src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+};
+#endif
+
+// Load mask?
+void func_801229FC(Player* player) {
+    if (player->maskObjectLoading == 1) {
+        // TODO: check if player->maskId is unsigned
+        s16 objectId = D_801BFDA0[(u8)player->maskId - 1];
+
+        osCreateMesgQueue(&player->maskObjectLoadQueue, &player->maskObjectLoadMsg, 1);
+        DmaMgr_SendRequestImpl(&player->maskDmaRequest, player->maskObjectSegment, gObjectTable[objectId].vromStart,
+                               gObjectTable[objectId].vromEnd - gObjectTable[objectId].vromStart, 0,
+                               &player->maskObjectLoadQueue, NULL);
+        player->maskObjectLoading++;
+    } else if (player->maskObjectLoading == 2) {
+        if (osRecvMesg(&player->maskObjectLoadQueue, NULL, OS_MESG_NOBLOCK) == 0) {
+            player->maskObjectLoading = 0;
+
+            if (player->currentMask == PLAYER_MASK_GREAT_FAIRY) {
+                s32 i;
+
+                for (i = 0; i < ARRAY_COUNT(D_801F58B0); i++) {
+                    func_80127B64(D_801F58B0[i], ARRAY_COUNT(D_801F58B0[i]), &player->bodyPartsPos[7]);
+                }
+            }
+        }
+    } else if ((player->currentMask != PLAYER_MASK_NONE) && (player->currentMask != (u8)player->maskId)) {
+        player->maskObjectLoading = 1;
+        player->maskId = player->currentMask;
+    } else if (player->currentMask == PLAYER_MASK_CIRCUS_LEADER) {
+        s32 i;
+
+        for (i = 0; i < ARRAY_COUNT(D_801F59C8); i++) {
+            D_801F59C8[i] += Rand_S16Offset(4, 23) + (s32)(fabsf(player->linearVelocity) * 50.0f);
+        }
+    }
+}
+
+void func_80122BA4(GlobalContext* globalCtx, struct_80122D44_arg1* arg1, s32 arg2, s32 alpha) {
+    if (arg2 == arg1->unk_00) {
+        s32 index;
+
+        arg1->unk_01--;
+        if (arg1->unk_01 < 0) {
+            arg1->unk_01 = 3;
+        }
+
+        index = arg1->unk_01;
+        arg1->unk_04[index].unk_00 = arg1->unk_00;
+        arg1->unk_04[index].unk_01 = alpha;
+        Matrix_CopyCurrentState(&arg1->unk_04[index].unk_04);
+
+        arg1->unk_00 = 0;
+    }
+}
+
+void func_80122C20(GlobalContext* globalCtx, struct_80122D44_arg1* arg1) {
+    struct_80122D44_arg1_unk_04* temp_v1 = &arg1->unk_04[0];
+    s32 phi_a1;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(arg1->unk_04); i++, temp_v1++) {
+        // Can't be `temp_v1->unk_01 != 0`
+        if (temp_v1->unk_01) {
+            phi_a1 = temp_v1->unk_00 == 3 ? 0x55 : 0x33;
+            if (phi_a1 >= temp_v1->unk_01) {
+                temp_v1->unk_01 = 0;
+            } else {
+                temp_v1->unk_01 -= phi_a1;
+            }
+        }
+    }
+}
+
+typedef struct {
+    /* 0x0 */ Color_RGB8 color;
+    /* 0x4 */ Gfx* dlist;
+} struct_801BFDD0; // size = 0x08
+
+extern struct_801BFDD0 D_801BFDD0[];
+#if 0
+struct_801BFDD0 D_801BFDD0[] = {
+    { { 180, 200, 255 }, object_link_goron_DL_00BDD8 },
+    { { 155, 0, 0 }, object_link_goron_DL_014690 },
+    { { 255, 0, 0 }, object_link_goron_DL_011AB8 },
+};
```

Comment:
Any idea what these are? Would probably explain the next function. My guess would be punch or spikes or something.

Reply:
object_link_goron_DL_00BDD8
![Screenshot from 2022-02-22 01-51-00](https://user-images.githubusercontent.com/7416381/155065320-91fe1c4c-b790-4063-a796-39d45f294b92.png)
object_link_goron_DL_014690 (this one uses an AnimatedMaterial)
![Screenshot from 2022-02-22 01-51-31](https://user-images.githubusercontent.com/7416381/155065323-bdd7a5f9-f28b-409e-91f3-2cf9183966be.png)
object_link_goron_DL_011AB8
![Screenshot from 2022-02-22 01-52-02](https://user-images.githubusercontent.com/7416381/155065331-8810f37c-9a5c-4809-98e7-379e654d8a27.png)

Reply:
At least the first two are obvious. Perhaps we should get a @tom-overton smiley face on the third...

Reply:
Your wish is my command; it indeed appears to be the punch:
![GLideN64_ZELDA_MAJORA'S_MASK_005](https://user-images.githubusercontent.com/12245827/155090688-060fe85a-c7b9-4dc1-aae7-81466c7b3610.png)

Reply:
Nice thanks.
So, what about naming them something like this?:
- `gLinkGoronCrouchingDL`
- `gLinkGoronRollingDL`
- `gLinkGoronPunchDL` / `gLinkGoronPunchEffectDL`

Reply:
I'm not sure if we prefer "LinkGoron" or "GoronLink", but those all sound pretty sensible to me.

Reply:
I think `Curled`, `RollingSpikesAndEffect` and `GoronPunchEffect`: it seems important to me that the latter two are labelled as effects. I assume that when rolling with spikes it just shows the squashed first one inside the second.